core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -70,21 +77,30 @@ class LogSegment(val log: FileMessageSet,
    *
    * It is assumed this method is being called from within a lock.
    *
-   * @param offset The first offset in the message set.
+   * @param firstOffset The first offset in the message set.
+   * @param lastOffset The last offset in the message set.
+   * @param timestamp The largest timestamp in the message set.
    * @param messages The messages to append.
    */
   @nonthreadsafe
-  def append(offset: Long, messages: ByteBufferMessageSet) {
+  def append(firstOffset: Long, lastOffset: Long, timestamp: Long, messages: ByteBufferMessageSet): Unit = {
```

Comment:
Yeah, Kafka currently deviates from that convention. I prefer the Scala coding style guide way, but there isn't a strong reason either way (one way is a little more consistent, but is more verbose). So, it probably makes sense to stick with Kafka's way for now as it would create a lot of diffs to change.